lib/letters.ml

```diff
@@ -1,14 +1,38 @@
-type config = {
-  sender : string;
-  username : string;
-  password : string;
-  hostname : string;
-  port : int option;
-  with_starttls : bool;
-  ca_dir : string;
-}
-
-type body = Plain of string | Html of string | Mixed of string * string * (string option)
+module Config = struct
+  type ca_certs = CA_bundle of string | CA_path of string | Detect
```

Comment:
Please name these `Ca_bundle` and `Ca_path`

Reply:
Nope, CA is a short and it is common practise not to drop the case like you asked. Also this is inline with X509 library's conventions.

Reply:
It is not idiomatic if you look at other libs: https://ocsigen.org/tyxml/4.4.0/manual/intro
`Html_sig`, `Html_types`, `Svg_sig`, `Svg_types`. 
I have not seen this kind of naming in the wild in OCaml. I have seen it in Java world (XMLWriter) which arguably defeats the purpose of CamelCasing.

Reply:
Yes, that's why I wrote this. We have stuff like `Ca_file` in the code that is coming from mirage.

Reply:
Very common practise is to keep all abbreviations in all caps when you have 1-3 char abbreviation. Also, in `ocaml-x509`, they follow this practise (not sure what would thay do if they had 4 char abbreviation or longer, but it does not matter here). See: [x509.mli](https://github.com/mirleft/ocaml-x509/blob/master/lib/x509.mli)

It can be argued if this defeats CamelCasing since abbreviations are a bit special case. Keeping abbreviations in all uppercase up to 3 chars is not just Java convention, it is a bit older one and this topic has been argued in different forums as long as CamelCase as a concept has existed. Claiming that idiomatic OCaml is to follow one or another is a bit wrong claim since the both are used. Because having both letters in uppercase is what X509 library does, I would keep both in uppercase here too since this part is related to the use of X509 lib.

Reply:
See more about this topic in [stackoverflow](https://stackoverflow.com/questions/15526107/acronyms-in-camelcase)

Reply:
And by the way, this is not even CamelCase that OCaml is using here.

Reply:
My example was Java specific to highlight this never ending discussion. Just saying that given the OCaml I have been reading so far, `CA_file` looks odd in my eyes. I suggest to follow conventions of Jane Street packages.

I have no strong feelings about it as long as we are consistent across our packages.

Reply:
The x509.mli seems to be a bad example of consistent naming.
For variants they have stuff like `EDI_party`, `DNS`, `Rfc_822`, `Ipsec_end`

Reply:
I see no value in using `CA_file` here and something `Html_parser` in some other package. So what is your rule? 2-3 letter abbreviations are all capital? This is less simple than just saying "first letter is capital", doesn't matter if abbreviation or not.

Reply:
(co-author of X509 here) yes, I agree that its naming is inconsistent in X509. unfortunately renaming has a cost (including bumping the major version and fixing reverse dependencies). I'd go with `Ca_file`, `Ca_bundle`, etc. here. (guidance for choosing identifiers in OCaml: https://ocaml.org/learn/tutorials/guidelines.html#How-to-choose-identifiers)

Reply:
@hannsem Thanks a lot for your work on X509 and your input! I guess the fact that the longest discussion in this review is about naming speaks for itself :smile:

Reply:
It seems I'm overruled 1 to 3 and I need to submit to the majority opinion :smile: 

Fixing.

Reply:
Fixed. I also change use of `bundle` to `cert` so that it is more consistent with rest of the code. So `CA_path` is now `Ca_path` and `CA_bundle` is `Ca_cert`.

Reply:
Thanks!